tfx_bsl/public/beam/run_inference.py

```diff
@@ -44,19 +53,84 @@ def RunInference(  # pylint: disable=invalid-name
+  converter = tf_example_record.TFExampleBeamRecord(
+    physical_format="inmem",
+    telemetry_descriptors=[],
+    schema=schema,
+    raw_record_column_name=_RECORDBATCH_COLUMN)
+
+  tensor_adapter_config = None
+  if schema:
+    tfxio = test_util.InMemoryTFExampleRecord(
+      schema=schema, raw_record_column_name=_RECORDBATCH_COLUMN)
+    tensor_adapter_config = tensor_adapter.TensorAdapterConfig(
+      arrow_schema=tfxio.ArrowSchema(),
+      tensor_representations=tfxio.TensorRepresentations())
+
+  return (examples
+          | 'ParseExamples' >> beam.Map(lambda example: example.SerializeToString())
+          | 'ConvertToRecordBatch' >> converter.BeamSource()
+          | 'RunInferenceImpl' >> run_inference.RunInferenceImpl(
+                  inference_spec_type, data_type,
+                  tensor_adapter_config=tensor_adapter_config))
+
+
+@beam.ptransform_fn
+@beam.typehints.with_input_types(tf.train.SequenceExample)
+@beam.typehints.with_output_types(prediction_log_pb2.PredictionLog)
+def RunInferenceOnSequenceExamples(  # pylint: disable=invalid-name
+    examples: beam.pvalue.PCollection,
+    inference_spec_type: model_spec_pb2.InferenceSpecType,
+    schema: Optional[schema_pb2.Schema] = None
+) -> beam.pvalue.PCollection:
+  """Run inference with a model.
+
+   There are two types of inference you can perform using this PTransform:
+   1. In-process inference from a SavedModel instance. Used when
+     `saved_model_spec` field is set in `inference_spec_type`.
+   2. Remote inference by using a service endpoint. Used when
+     `ai_platform_prediction_model_spec` field is set in
+     `inference_spec_type`.
+
+  Args:
+    examples: A PCollection containing sequence examples.
+    inference_spec_type: Model inference endpoint.
+    Schema [optional]: required for models that requires
+      multi-tensor inputs.
 
   Returns:
     A PCollection containing prediction logs.
   """
 
-  return (
-      examples |
-      'RunInferenceImpl' >> run_inference.RunInferenceImpl(inference_spec_type))
+  data_type = DataType.SEQUENCEEXAMPLE
+  converter = tf_sequence_example_record.TFSequenceExampleBeamRecord(
+    physical_format="inmem",
+    telemetry_descriptors=[],
+    schema=schema,
+    raw_record_column_name=_RECORDBATCH_COLUMN)
+
+  tensor_adapter_config = None
+  if schema:
+    tfxio = test_util.InMemoryTFExampleRecord(
+      schema=schema, raw_record_column_name=_RECORDBATCH_COLUMN)
+    tensor_adapter_config = tensor_adapter.TensorAdapterConfig(
+      arrow_schema=tfxio.ArrowSchema(),
+      tensor_representations=tfxio.TensorRepresentations())
+
+  return (examples
+          | 'ParseExamples' >> beam.Map(lambda example: example.SerializeToString())
+          | 'ConvertToRecordBatch' >> converter.BeamSource()
+          | 'RunInferenceImpl' >> run_inference.RunInferenceImpl(
+                  inference_spec_type, data_type,
+                  tensor_adapter_config=tensor_adapter_config))
```

Comment:
Same here, can this be inside of run_inference.py, so that we only expose 1 ptransform here?